src/pages/storage/forms/StoragePoolClusteredSourceSelector.tsx

```diff
@@ -0,0 +1,41 @@
+import { Label } from "@canonical/react-components";
+import ClusterSpecificInput from "components/forms/ClusterSpecificInput";
+import { FormikProps } from "formik";
+import { FC } from "react";
+import { ensureEditMode } from "util/instanceEdit";
+import { StoragePoolFormValues } from "./StoragePoolForm";
+import { useClusterMembers } from "context/useClusterMembers";
+import { focusField } from "util/formFields";
+
+interface Props {
+  formik: FormikProps<StoragePoolFormValues>;
+}
+
+const StoragePoolClusteredSourceSelector: FC<Props> = ({ formik }) => {
+  const { data: clusterMembers = [] } = useClusterMembers();
+  const memberNames: string[] = [];
+  clusterMembers.forEach((member) => memberNames.push(member.server_name));
+
+  return (
+    <>
+      <Label>Source</Label>
```

Comment:
```suggestion
      <Label forId="sourcePerClusterMember">Source</Label>
```

We should also make this field required if we are creating the storage pool